### PR TITLE
property: make cache flushing slightly less deterministic

### PR DESCRIPTION
**Pull request by paulidale**
If there is no timer available to seed the stochastic flushing, revert to a global seed that gets updated each flush.
This is mitigating the possibility that the cache flushing could be manipulated by an attacker to cause a DoS.
Such an attack is purely theoretical at the moment.

- [ ] documentation is added or updated
- [ ] tests are added or updated

### CODE REVIEW

**Comment by t8m** [comment]
I.E., is this a bug fix or a feature? I am undecided.

File: crypto/property/property.c [hunk 1]
```diff
@@ -649,13 +651,21 @@ static void impl_cache_flush_one_alg(ossl_uintmax_t idx, ALGORITHM *alg,
...
+        /* If there is no timer available, seed another way */
+        state.using_global_seed = 1;
+        state.seed = tsan_load(&global_seed);
+    }
     store->cache_need_flush = 0;
     ossl_sa_ALGORITHM_doall_arg(store->algs, &impl_cache_flush_one_alg, &state);
     store->cache_nelem = state.nelem;
+    /* Without a timer, update the global seed */
+    if (state.using_global_seed)
+        tsan_add(&global_seed, state.seed);
```
Comment (mattcaswell) [inline crypto/property/property.c:668]: This seems to just double the global_seed every time (barring multiple threads racing with each other). So with a `uint32_t` that only gives 32 possible seed values. So, only, very slightly less deterministic. Wouldn't a simple increment be better?

Comment (paulidale) [reply crypto/property/property.c:668]: It doesn't do anything remotely like that: state.seed is updated many times on line 633 as part of the doall call.  This continues the pseudo random sequence from where it left off.  At least single threaded...

Comment (mattcaswell) [reply crypto/property/property.c:668]: Ah! the update on line 633 was the bit I was missing.Thanks.

**Review by mattcaswell — APPROVED** [review]

**Comment by mattcaswell** [comment]
Pushed to master.

I'll leave this PR open while the question about backport above is unresolved.

**Comment by paulidale** [comment]
+1 for a bug fix.

**Comment by paulidale** [comment]
Bug or feature?

**Comment by t8m** [comment]
I am OK with considering this a bug fix. @mattcaswell ?

**Comment by paulidale** [comment]
Merged to 3.0.

**Pull request closed without merging**